spring-cloud-alibaba-sentinel/src/main/java/org/springframework/cloud/alibaba/sentinel/endpoint/SentinelHealthIndicator.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * If Dashboard and DataSource are both OK, the health status is UP.
+ *</p>
+ *
+ * <p>
+ * Note:
+ * If Sentinel isn't enabled, the health status is up.
+ * If Sentinel Dashboard isn't configured, it's OK and mark the status of Dashboard with UNKNOWN.
+ * More informations are provided in details.
+ * </p>
+ *
+ * @author cdfive
+ */
+public class SentinelHealthIndicator extends AbstractHealthIndicator {
+
+    private DefaultListableBeanFactory beanFactory;
+
+    private SentinelProperties sentinelProperties;
+
+    public SentinelHealthIndicator(DefaultListableBeanFactory beanFactory, SentinelProperties sentinelProperties) {
+        this.beanFactory = beanFactory;
+        this.sentinelProperties = sentinelProperties;
+    }
+
+    @Override
+    protected void doHealthCheck(Health.Builder builder) throws Exception {
+        Map<String, Object> detailMap = new HashMap<>();
+
+        // If sentinel isn't enabled, set the status up and set the enabled to false in detail
+        if (!sentinelProperties.isEnabled()) {
+            detailMap.put("enabled", false);
+            builder.up().withDetails(detailMap);
+            return;
+        }
+
+        detailMap.put("enabled", true);
+
+        // Check health of Dashboard
+        boolean dashboardUp = true;
+        String consoleServer = TransportConfig.getConsoleServer();
+        if (StringUtils.isEmpty(consoleServer)) {
+            // If Dashboard isn't configured, mark the status of Dashboard with UNKNOWN and the dashboardUp is still true
+            detailMap.put("dashboard", new Status(Status.UNKNOWN.getCode(), "dashboard isn't configured"));
+        } else {
+            // If Dashboard is configured, send a heartbeat message to it and check the result
+            HeartbeatSender heartbeatSender = HeartbeatSenderProvider.getHeartbeatSender();
+            boolean result = heartbeatSender.sendHeartbeat();
+            if (result) {
+                detailMap.put("dashboard", Status.UP);
+            } else {
+                // If failed to send heartbeat message, means that the Dashboard is DOWN
+                dashboardUp = false;
+                detailMap.put("dashboard", new Status(Status.DOWN.getCode(), consoleServer + " can't be connected"));
+            }
+        }
+
+        // Check health of DataSource
+        boolean dataSourceUp = true;
+        Map<String, Object> dataSourceDetailMap = new HashMap<>();
+        detailMap.put("dataSource", dataSourceDetailMap);
+
+        // Get all DataSources and each call loadConfig to check if it's OK
+        Map<String, AbstractDataSource> dataSourceMap = beanFactory.getBeansOfType(AbstractDataSource.class);
+        for (Map.Entry<String, AbstractDataSource> dataSourceMapEntry : dataSourceMap.entrySet()) {
+            String dataSourceBeanName = dataSourceMapEntry.getKey();
+            AbstractDataSource dataSource = dataSourceMapEntry.getValue();
+            try {
+                dataSource.loadConfig();
```

Comment:
Even if the dynamic config center is down, the `loadConfig()` might return successfully (e.g. for Nacos client, it might retrieve from the local cache). But in most circumstances it's okay.

Reply:
+1 for this. I think is ok.

Reply:
Tried with NacosDataSource, when stop the Nacos Server, the `loadConfig()` indeed return the rules which are added previously. To handle this,  maybe Nacos client need modify or a separate health check method need to be added in `AbstractDataSource` interface.  I also add a Note comment in file. Thanks for reviewing!

Reply:
👍